Mathlib/Analysis/NormedSpace/BanachSteinhaus.lean

```diff
@@ -100,33 +58,9 @@ open Filter
 /-- Given a *sequence* of continuous linear maps which converges pointwise and for which the
 domain is complete, the Banach-Steinhaus theorem is used to guarantee that the limit map
 is a *continuous* linear map as well. -/
-def continuousLinearMapOfTendsto [CompleteSpace E] [T2Space F] (g : ℕ → E →SL[σ₁₂] F) {f : E → F}
-    (h : Tendsto (fun n x => g n x) atTop (𝓝 f)) : E →SL[σ₁₂] F where
-  toFun := f
-  map_add' := (linearMapOfTendsto _ _ h).map_add'
-  map_smul' := (linearMapOfTendsto _ _ h).map_smul'
-  cont := by
-    -- show that the maps are pointwise bounded and apply `banach_steinhaus`
-    have h_point_bdd : ∀ x : E, ∃ C : ℝ, ∀ n : ℕ, ‖g n x‖ ≤ C := by
-      intro x
-      rcases cauchySeq_bdd (tendsto_pi_nhds.mp h x).cauchySeq with ⟨C, -, hC⟩
-      refine' ⟨C + ‖g 0 x‖, fun n => _⟩
-      simp_rw [dist_eq_norm] at hC
-      calc
-        ‖g n x‖ ≤ ‖g 0 x‖ + ‖g n x - g 0 x‖ := norm_le_insert' _ _
-        _ ≤ C + ‖g 0 x‖ := by linarith [hC n 0]
-    cases' banach_steinhaus h_point_bdd with C' hC'
-    /- show the uniform bound from `banach_steinhaus` is a norm bound of the limit map
-             by allowing "an `ε` of room." -/
-    refine'
-      AddMonoidHomClass.continuous_of_bound (linearMapOfTendsto _ _ h) C' fun x =>
-        le_of_forall_pos_lt_add fun ε ε_pos => _
-    cases' Metric.tendsto_atTop.mp (tendsto_pi_nhds.mp h x) ε ε_pos with n hn
-    have lt_ε : ‖g n x - f x‖ < ε := by
-      rw [← dist_eq_norm]
-      exact hn n (le_refl n)
-    calc
-      ‖f x‖ ≤ ‖g n x‖ + ‖g n x - f x‖ := norm_le_insert _ _
-      _ < ‖g n‖ * ‖x‖ + ε := by linarith [lt_ε, (g n).le_op_norm x]
-      _ ≤ C' * ‖x‖ + ε := by nlinarith [hC' n, norm_nonneg x]
+def continuousLinearMapOfTendsto [CompleteSpace E] [T2Space F] {l : Filter α}
+    [l.IsCountablyGenerated] [l.NeBot] (g : α → E →SL[σ₁₂] F) {f : E → F}
+    (h : Tendsto (fun n x ↦ g n x) l (𝓝 f)) :
+    E →SL[σ₁₂] F :=
+  (norm_withSeminorms 𝕜₂ F).continuousLinearMapOfTendsto g h
```

Comment:
I'm not sure we need this as a separate `def` anymore, and I would advocate against it, but if we are going to have it, perhaps it can be an `abbrev`.

Reply:
You mean keeping only the general definition right? Or do you mean we shouldn't express this using a def at all?

Assuming the first case, the reason I kept this one separate is because of the extra `WithSeminorms` argument. That's the downside of not using `LocallyConvexSpace`, we can't have the typeclass system take care of the generalization here. One solution would be to have a typeclass saying "there exists a family of seminorms generating the topology" but I don't really want to do that. Otherwise we could just expect the user to provide `norm_withSeminorms 𝕜₂ F`, but that sounds a bit annoying too. Anyway I'll definitely make it an `abbrev`.

Reply:
> I thought making the use supply norm_withSeminorms wasn't too burdensome.

It's definitely not a huge burden, but I think it's better if people can just work in normed spaces out of the box since it's such a common place.